src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -0,0 +1,386 @@
+                diagnostics.Add(node, useSiteDiagnostics);
+            }
+
+            if (convertedLeftType is null || convertedRightType is null)
+            {
+                convertedLeftType = convertedLeftType ?? leftType ?? CreateErrorType();
+                convertedRightType = convertedRightType ?? rightType ?? CreateErrorType();
+                hasErrors = true;
+            }
+
+            if (hasErrors)
+            {
+                ReportBinaryOperatorError(node, diagnostics, node.OperatorToken, left, right, resultKind);
+                //resultOperatorKind &= ~BinaryOperatorKind.TypeMask; // PROTOTYPE(tuple-equality) Not sure what this is for
+            }
+
+            ConvertToBool(returnType, node, kind, diagnostics, out Conversion boolConversion, out UnaryOperatorSignature boolOperator);
+            return new TupleBinaryOperatorInfo.Single(convertedLeftType, convertedRightType, resultOperatorKind, resultMethod, boolConversion, boolOperator);
+        }
+
+        private void ConvertToBool(TypeSymbol type, BinaryExpressionSyntax node, BinaryOperatorKind binaryOperator, DiagnosticBag diagnostics,
+            out Conversion boolConversion, out UnaryOperatorSignature boolOperator)
+        {
+            BoundExpression comparisonResult = new BoundTupleOperandPlaceholder(node, type);
+            var boolean = GetSpecialType(SpecialType.System_Boolean, diagnostics, node);
+
+            // Is the operand implicitly convertible to bool?
+
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            var conversion = this.Conversions.ClassifyConversionFromExpression(comparisonResult, boolean, ref useSiteDiagnostics);
+            diagnostics.Add(node, useSiteDiagnostics);
+
+            if (conversion.IsImplicit)
+            {
+                boolConversion = conversion;
+                boolOperator = default;
+                return;
+            }
+
+            // It was not. Does it implement operator true (or false)?
+
+            UnaryOperatorKind boolOpKind;
+            switch (binaryOperator)
+            {
+                case BinaryOperatorKind.Equal:
+                    boolOpKind = UnaryOperatorKind.False;
+                    break;
+                case BinaryOperatorKind.NotEqual:
+                    boolOpKind = UnaryOperatorKind.True;
+                    break;
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(binaryOperator);
+            }
+
+            LookupResultKind resultKind;
+            ImmutableArray<MethodSymbol> originalUserDefinedOperators;
+            var best = this.UnaryOperatorOverloadResolution(boolOpKind, comparisonResult, node, diagnostics, out resultKind, out originalUserDefinedOperators);
+            if (best.HasValue)
+            {
+                boolConversion = best.Conversion;
```

Comment:
Note: This logic follows the binding for `if` statement. But I don't think we ever get a useful conversion in the case where we use `true`/`false` operators. The conversion is always identity. 
That's affirmed by running our emit tests to see if we can get a non-identity conversion when binding `if` with a `true`/`false` operator.